src/hip/hip_build_utils.cpp

```diff
@@ -120,6 +120,7 @@ boost::filesystem::path HipBuild(boost::optional<TmpDir>& tmp_dir,
 
     params += " -Wno-unused-command-line-argument -I. ";
     params += MIOPEN_STRINGIZE(HIP_COMPILER_FLAGS);
+    params += " --" + std::string(MIOPEN_ARCH_FLAG) + "=" + dev_name;
```

Comment:
Move it to line 115 (i.e. replace line 115 with it). And what about line 116 ("--cuda-device-only")?